### PR TITLE
Added new server basic functionality

### PR DESCRIPTION
**Pull request by gdeepchadha**
- Added routes for listing and searching restaurants from the database
 - Added route to insert restaurant data into the database

### CODE REVIEW

**Review by GITANG1 — DISMISSED** [review]
remove the comments and add newline

**Review by nakhe93 — DISMISSED** [review]
Looks Good.

File: app.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+var bodyParser = require('body-parser');
+
+var index = require('./routes/index');
+var insertRestaurant = require('./routes/insertRestaurant');
+var restaurants = require('./routes/restaurants');
+var app = express();
+
+
+// uncomment after placing your favicon in /public
+//app.use(favicon(path.join(__dirname, 'public', 'favicon.ico')));
```
Comment (srishtihunjan) [inline app.js]: do we need these 2 commented lines?

File: bin/www [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+// http.createServer(app);
+
+/**
+ * Listen on provided port, on all network interfaces.
+ */
+
+app.listen(port,hostname);
+//server.on('error', onError);
+//server.on('listening', onListening);
+
```
Comment (srishtihunjan) [inline bin/www]: could you check if we need the express server to be listening to the requests? That is, can you uncomment these 2 lines?

File: models/restaurantSchema.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
+var mongoose = require('mongoose');
+var textSearch = require('mongoose-text-search');
+var Schema = mongoose.Schema;
+
+
+//dishes_schema.plugin(textSearch);
+
+// add a text index to the tags array
+//dishes_schema.index({ name: 'text' });
+
```
Comment (srishtihunjan) [inline models/restaurantSchema.js]: can we remove the commented code?

**Review by srishtihunjan — CHANGES_REQUESTED** [review]
Please could you remove the commented code that's not necessary?

**Review by srishtihunjan — APPROVED** [review]
Looks good

**Review by GITANG1 — APPROVED** [review]
this works!

**Review by nakhe93 — APPROVED** [review]
Looks good

**Pull request merged**